test/skimNtuple2018_HHbtag.cpp

```diff
@@ -1778,89 +1779,141 @@ int main (int argc, char** argv)
-	  // looking at all leptons in decay chain, but end up comparing the last one (e/mu/tau) or the mother for nu count
-	  // which one it is should not make much difference with DR<0.3
-	  vGenTauVis.SetPxPyPzE (theBigTree.genpart_px->at(igen),
-				 theBigTree.genpart_py->at(igen),
-				 theBigTree.genpart_pz->at(igen),
-				 theBigTree.genpart_e->at(igen));
-	  match1 = (vGenTauVis.DeltaR(tlv_firstLepton)<0.3);
-	  match2 = (vGenTauVis.DeltaR(tlv_secondLepton)<0.3);
-
-	  if( match1 && !match2 ) {
-	    vGenTau1 = vGenTauVis;
-	    gentau1_idx = mothIdx;
-	  }
-	  if( !match1 &&  match2 ) {
-	    vGenTau2 = vGenTauVis;
-	    gentau2_idx = mothIdx;
-	  }
-	  if( match1 &&  match2 ) { // unlikely I guess
-	    if (vGenTauVis.DeltaR(tlv_firstLepton)<vGenTauVis.DeltaR(tlv_secondLepton)){
-	      vGenTau1 = vGenTauVis;
-	      gentau1_idx = mothIdx;
-	    } else {
-	      vGenTau2 = vGenTauVis;
-	      gentau2_idx = mothIdx;
-	    }
-	  }
-	} // endif(isLepton)
-
-	TLorentzVector vGenNuVis;
-	if(isNeutrino){
-	  vGenNuVis.SetPxPyPzE (theBigTree.genpart_px->at(igen), theBigTree.genpart_py->at(igen), theBigTree.genpart_pz->at(igen), theBigTree.genpart_e->at(igen));
-
-	  //cout << "Found gen neutrino : id=" << igen << " | tauMothID=" <<theBigTree.genpart_TauMothInd->at(igen) << endl;
-
-	  if(theBigTree.genpart_TauMothInd->at(igen)==gentau1_idx) // neutrino comes from tau1
-	    vGenNu1+=vGenNuVis;
-	  else if(theBigTree.genpart_TauMothInd->at(igen)==gentau2_idx) // neutrino comes from tau2
-	    vGenNu2+=vGenNuVis;
-	  else // neutrino is unmatched
-	    vGenNuNoMatch+=vGenNuVis;
-	}
-      } //end loop on gen part
-    } // endif(isMC)
-      // -- END GEN NEUTRINO DEFINITION
-      /////////////////////////////////
+        // -- GEN NEUTRINO, LEPTON, B-QUARK DEFINITION:
+        // -> Adding gen-matched info for the 2 taus neutrinos for tauTau ID training tests
+        TLorentzVector vGenNu1; // neutrino associated to tau1
+        TLorentzVector vGenNu2; // neutrino associated to tau2
+        TLorentzVector vGenNuNoMatch; // neutrino associated to tau2
+        // also getting gen vis tau 4vec for reco matching
+        TLorentzVector vGenTau1Vis;
+        TLorentzVector vGenTau2Vis;
+        int gentau1_idx = -1;
+        int gentau2_idx = -1;
+
+        std::vector<TLorentzVector> vGenLeptons;
+        std::vector<Int_t> vGenLeptonPdgIds;
+        std::vector<Int_t> vGenLeptonMotherPdgIds;
+
+        std::vector<TLorentzVector> vGenBQuarks;
+        std::vector<Int_t> vGenBQuarkMotherPdgIds;
+        std::vector<Int_t> vGenBQuarkStatus;
+
+        if(isMC) {
+          for (unsigned int igen = 0; igen < theBigTree.genpart_px->size(); igen++) {
+            // only looking at gen e/mu/tau and neutrinos
+            int pdg = fabs(theBigTree.genpart_pdg->at(igen));
+
+            if(pdg == 5 && CheckBit(theBigTree.genpart_flags->at(igen), 8) && CheckBit(theBigTree.genpart_flags->at(igen), 12) && theBigTree.genpart_status->at(igen) != 21){
```

Comment:
could you add a comment stating what the flag bits 8 and 12 point to? I think it can be found elsewhere in the loop but that would avoid having to look for them every time

Reply:
done